src/emulate.c

```diff
@@ -370,6 +370,8 @@ static bool has_loops = false;
 
 #if RV32_HAS(SYSTEM) && !RV32_HAS(ELF_LOADER)
 extern void emu_update_uart_interrupts(riscv_t *rv);
+extern void emu_update_rtc_interrupts(riscv_t *rv);
+extern uint64_t get_now_nsec(rtc_t *rtc);
```

Comment:
Since this function is used to obtain the current precise time, I assume it is performance critical (please correct me if I'm wrong). The function itself is also quite short, so IMHO it would be a good candidate to move into a header as an inline function. Even if it is not made inline, since it is called from both emulate.c and rtc, I would still prefer to place the prototype in a suitable header rather than relying on extern.